test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -198,18 +128,21 @@ TEST_P(ProxyFilterCircuitBreakerIntegrationTest, Basic) {
       {":authority",
        fmt::format("localhost:{}", fake_upstreams_[0]->localAddress()->ip()->port())}};
 
-  auto response = codec_client_->makeRequestWithBody(request_headers, 1024);
-  response->waitForEndStream();
-  EXPECT_EQ(1, test_server_->gauge("dns_cache.foo.circuit_breakers.rq_pending_open"));
-  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.dns_rq_pending_overflow")->value());
+  auto response =
+      sendRequestAndWaitForResponse(request_headers, 1024, default_response_headers_, 1024);
+  checkSimpleRequestSuccess(1024, 1024, response.get());
+  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.dns_query_attempt")->value());
+  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.host_added")->value());
 
-  EXPECT_TRUE(response->complete());
-  EXPECT_EQ("503", response->headers().Status()->value().getStringView());
+  // Now send another request. This should hit the DNS cache.
+  response = sendRequestAndWaitForResponse(request_headers, 512, default_response_headers_, 512);
+  checkSimpleRequestSuccess(512, 512, response.get());
+  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.dns_query_attempt")->value());
+  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.host_added")->value());
 }
 
-// A basic test where we pause a request to lookup localhost, and then do another request which
-// should hit the TLS cache.
-TEST_P(ProxyFilterIntegrationTest, RequestWithBody) {
+TEST_P(ProxyFilterIntegrationTest, HeaderRequestWithClusterCircuitBreaker) {
```

Comment:
This is not a `HeaderRequest` but `RequestWithBody`?

Reply:
done